deploy.py

```diff
@@ -98,27 +95,62 @@ def create_control_node(config):
             'CAPABILITY_NAMED_IAM'
         ],
     )
-    print("deployed stack!")
+    print("Uploaded deployment spec. CloudFormation will take it from here.")
+
+
+def upload_ssh_key(key_name):
+    """Wait for the secret to be created in the stack, then update it."""
+    id = "{}.pem".format(key_name)
+    while True:
+        try:
+            secrets.describe_secret(SecretId=id)
+            break
+        except:
+            time.sleep(0.25)
+    secrets.update_secret(SecretId=id, SecretString=open(id, 'r').read())
+    print("Uploaded SSH key.")
+
+
+def fail(msg):
+    """Print and quit."""
+    print(msg)
+    sys.exit(1)
+
 
 if __name__ == "__main__":
 
     # The only argument required is the tag that makes this deployment and all its resources unique.
     parser = argparse.ArgumentParser()
-    parser.add_argument("--tag", "-t", required=False, help="tag to build, must be alphanumeric like \"prod\" or \"test\"")
+    parser.add_argument("--tag", "-t",
+                        required=False,
+                        default="test",
+                        help="tag to build, must be alphanumeric like \"prod\" or \"test\"")
+
+    parser.add_argument("--project", "-p",
+                        required=False,
+                        default="umsi-easy-hub",
+                        help="name of project, used in all AWS resources")
+
+    parser.add_argument("--wait", "-w",
+                        required=False,
+                        default=False,
+                        action='store_true',
+                        help="wait until the control node has completed and display it's public IP address")
 
     args = parser.parse_args()
 
-    # Default tag is "test"
-    if args.tag is None:
-        tag = "test"
-    else:
-        tag = args.tag
+    if args.project != "umsi-easy-hub":
+        fail("Using a different project name is currently unsupported.")
```

Comment:
Do we really want to not support other project names when we are thinking around other people besides MADS using EasyHub?

Reply:
We do, which is why I added the flag. It will just require a lot of work. This project name is hardcoded all over.

Reply:
OK, I guess the next step (not in this PR, in the future) is removing all the hard-coded pieces and get rid of the flag :wink:

Reply:
https://github.com/umsi-mads/main/issues/371